programs/zk-elgamal-proof/src/lib.rs

```diff
@@ -0,0 +1,317 @@
+            ic_msg!(invoke_context, "VerifyZeroCiphertext");
+            process_verify_proof::<ZeroCiphertextProofData, ZeroCiphertextProofContext>(
+                invoke_context,
+            )
+        }
+        ProofInstruction::VerifyCiphertextCiphertextEquality => {
+            invoke_context
+                .consume_checked(VERIFY_CIPHERTEXT_CIPHERTEXT_EQUALITY_COMPUTE_UNITS)
+                .map_err(|_| InstructionError::ComputationalBudgetExceeded)?;
+            ic_msg!(invoke_context, "VerifyCiphertextCiphertextEquality");
+            process_verify_proof::<
+                CiphertextCiphertextEqualityProofData,
+                CiphertextCiphertextEqualityProofContext,
+            >(invoke_context)
+        }
+        ProofInstruction::VerifyCiphertextCommitmentEquality => {
+            invoke_context
+                .consume_checked(VERIFY_CIPHERTEXT_COMMITMENT_EQUALITY_COMPUTE_UNITS)
+                .map_err(|_| InstructionError::ComputationalBudgetExceeded)?;
+            ic_msg!(invoke_context, "VerifyCiphertextCommitmentEquality");
+            process_verify_proof::<
+                CiphertextCommitmentEqualityProofData,
+                CiphertextCommitmentEqualityProofContext,
+            >(invoke_context)
+        }
+        ProofInstruction::VerifyPubkeyValidity => {
+            invoke_context
+                .consume_checked(VERIFY_PUBKEY_VALIDITY_COMPUTE_UNITS)
+                .map_err(|_| InstructionError::ComputationalBudgetExceeded)?;
+            ic_msg!(invoke_context, "VerifyPubkeyValidity");
+            process_verify_proof::<PubkeyValidityProofData, PubkeyValidityProofContext>(
+                invoke_context,
+            )
+        }
+        ProofInstruction::VerifyPercentageWithCap => {
+            invoke_context
+                .consume_checked(VERIFY_PERCENTAGE_WITH_CAP_COMPUTE_UNITS)
+                .map_err(|_| InstructionError::ComputationalBudgetExceeded)?;
+            ic_msg!(invoke_context, "VerifyPercentageWithCap");
+            process_verify_proof::<PercentageWithCapProofData, PercentageWithCapProofContext>(
+                invoke_context,
+            )
+        }
+        ProofInstruction::VerifyBatchedRangeProofU64 => {
+            invoke_context
+                .consume_checked(VERIFY_BATCHED_RANGE_PROOF_U64_COMPUTE_UNITS)
+                .map_err(|_| InstructionError::ComputationalBudgetExceeded)?;
+            ic_msg!(invoke_context, "VerifyBatchedRangeProof64");
+            process_verify_proof::<BatchedRangeProofU64Data, BatchedRangeProofContext>(
+                invoke_context,
+            )
+        }
+        ProofInstruction::VerifyBatchedRangeProofU128 => {
+            invoke_context
+                .consume_checked(VERIFY_BATCHED_RANGE_PROOF_U128_COMPUTE_UNITS)
+                .map_err(|_| InstructionError::ComputationalBudgetExceeded)?;
+            ic_msg!(invoke_context, "VerifyBatchedRangeProof128");
```

Comment:
micro-nit if you want
```suggestion
            ic_msg!(invoke_context, "VerifyBatchedRangeProofU128");
```